content/app/algo/three/traversing.js

```diff
@@ -0,0 +1,61 @@
+define("content/app/algo/three/traversing", function(require, exports, module) {
+
+	/**
+	 * Utility for traversing three structure.
+	 * It is type independent. Visitor should be
+	 * implemented for specific type.
+	 */
+	var Traversing = Class(Object, {
+
+		$statics : {
+
+			/**
+			 * Traverse element with provided visitor.
+			 * @param  {Object} 			element
+			 * @param  {AbstractVisitor} 	visitor
+			 */
+			traverse : function (element, visitor) {
+				if (element) {
+					visitor.visit(element);
+					var children = visitor.extractChildren(element);
```

Comment:
I think that definitely extract children should not be part of visitor, but instead part of traversing. Visitor should not know anything about the structure, it should be dumb and know only to visit single element. If you wanted to use one traversing for everything and use visitor as a way to extract children, then this is no visitor pattern. You should make base class for traversing, and override default _getChildren in subclass.

Also, it is little bit spaghetti code that traverse is calling traverseList and traverseList is calling traverse. Just put everything in one method.